contracts/Insurance.sol

```diff
@@ -15,122 +14,112 @@ contract Insurance is IInsurance, Ownable, SafetyWithdraw {
     using LibMath for uint256;
     using LibMath for int256;
 
-    int256 public override constant INSURANCE_MUL_FACTOR = 1000000000;
+    int256 public constant override INSURANCE_MUL_FACTOR = 1000000000;
     uint256 public constant SAFE_TOKEN_MULTIPLY = 1e18;
-    address public immutable TCRTokenAddress;
-    IAccount public account;
     ITracerPerpetualsFactory public perpsFactory;
 
-    struct StakePool { 
-        address market;
-        address collateralAsset;
-        uint256 amount; // amount of underlying collateral in pool
-        address token; // tokenized holdings of pool - not necessarily 1 to 1 with underlying
-        mapping(address => uint256) userDebt; // record of user debt to the pool
-        mapping(address => uint256) lastRewardsUpdate;
-    }
+    address public collateralAsset; // Address of collateral asset
+    uint256 public poolAmount; // amount of underlying collateral in pool
+    address public token; // tokenized holdings of pool - not necessarily 1 to 1 with underlying
 
-    // Tracer market => supported or not supported
-    mapping(address => bool) internal supportedTracers;
-    // Tracer market => StakePool
-    mapping(address => StakePool) internal pools;
+    ITracerPerpetualSwaps public tracer; // Tracer associated with Insurance Pool
 
     event InsuranceDeposit(address indexed market, address indexed user, uint256 indexed amount);
     event InsuranceWithdraw(address indexed market, address indexed user, uint256 indexed amount);
     event InsurancePoolDeployed(address indexed market, address indexed asset);
-    event InsurancePoolRewarded(address indexed market, uint256 indexed amount);
 
-    constructor(address TCR) public {
-        TCRTokenAddress = TCR;
+    constructor(address _tracer) {
+        require(perpsFactory.validTracers(_tracer), "Pool not deployed by perpsFactory");
+ 
+        tracer = ITracerPerpetualSwaps(_tracer);
+        InsurancePoolToken _token = new InsurancePoolToken("Tracer Pool Token", "TPT");
+        token = address(_token);
+        collateralAsset = tracer.tracerBaseToken();
+        poolAmount = 0;
+
+        // emit InsurancePoolDeployed(_tracer, tracer.tracerBaseToken());
```

Comment:
May need to check with @dospore what affect this change will have on the frontend. I imagine he is watching for all events from the old insurance contract. We should probably still emit this event IMO

Reply:
Oversight on my part; fixed now